src/app/submissions/submissions.component.ts

```diff
@@ -69,7 +69,7 @@ export class SubmissionsComponent implements OnInit, AfterViewChecked, OnDestroy
 
   ngOnInit() {
     this.setMode();
-    this.filter.type = this.route.snapshot.paramMap.get('type') || 'exam';
+    this.onFilterChange(this.filter.type, 'type');
```

Comment:
The filter should be set to `this.route.snapshop.paramMap.get('type') || 'exam'`, not itself.